misk-core/src/main/kotlin/misk/queuing/TimedBlockingQueue.kt

```diff
@@ -0,0 +1,171 @@
+package misk.queuing
+
+import java.time.Duration
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.TimeUnit
+
+/**
+ * This BlockingQueue implementation allows measuring how long dequeued items have been
+ * in the queue. It requires a delayHandler which consumes the latency every time an item is
+ * removed from the queue.
+ * Note that the the actual removal of an item from the queue and the delayHandler invocation
+ * are not atomic, hence it may not be appropriate for applications which require
+ * a higher level of accuracy.
+ * @param <T> The type of the elements held in this queue
+ */
+class TimedBlockingQueue<T>(
```

Comment:
internal

Reply:
This is being used by one of the modules in `misk.web`. Adding internal here causes a compilation error unless we move this class to the same package as the module. I feel that putting this class in a separate package and making it public allows reuse.

Reply:
Wanna move it to the misk.web module and keep it in this package? note that internal is module-scoped and is unrelated to the package name.

Reply:
Done.